api/tests/routes/backoffice/accounts_test.py

```diff
@@ -2403,3 +2412,37 @@ def test_get_tunnel(self):
         users_factories.DepositGrantFactory(user=user)
         tunnel_end = _get_tunnel(user, eligibility_history, fraud_reviews_desc)
         assert tunnel_end["progress"] == 100
+
+
+class AnonymizePublicAccountTest(PostEndpointHelper):
+    endpoint = "backoffice_web.public_accounts.anonymize_public_account"
+    endpoint_kwargs = {"user_id": 1}
+    needed_permission = perm_models.Permissions.ANONYMIZE_PUBLIC_ACCOUNT
+
+    def test_anonymize_public_account(
+        self,
+        legit_user,
+        authenticated_client,
+    ):
+        user = users_factories.UserFactory()
+
+        response = self.post_to_endpoint(authenticated_client, user_id=user.id)
+        assert response.status_code == 302
+
+        expected_url = url_for("backoffice_web.public_accounts.get_public_account", user_id=user.id, _external=True)
+        assert response.location == expected_url
+
+        assert "Anonymous" in user.firstName
+        assert user.roles == [users_models.UserRole.ANONYMIZED]
+        history = history_models.ActionHistory.query.one_or_none()
+        assert history.actionType == history_models.ActionType.USER_ANONYMIZED
```

Comment:
Comme je te l'avais dit, il est important de s'assurer que l'auteur est bien tracé : 
```suggestion
        assert history.actionType == history_models.ActionType.USER_ANONYMIZED
        assert history.authorUser == legit_user
        assert history.user == user
```

Reply:
```
assert history.authorUser == legit_user
        assert history.user == user
```
FAIT